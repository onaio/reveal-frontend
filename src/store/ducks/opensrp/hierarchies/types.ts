import { Dictionary } from '@onaio/utils';
import TreeModel from 'tree-model';

/** describes a node's attribute field */
export interface HierarchySingleNodeAttributes {
  geographicLevel: number;
  structureCount?: number;
}

/** Generic type to create types for a single node where children prop is generic */
export interface HierarchySingleNode<TChild> {
  id: string;
  label: string;
  node: {
    locationId: string;
    name: string;
    parentLocation?: { locationId: string; voided: boolean };
    attributes: HierarchySingleNodeAttributes;
    voided: boolean;
  };
  children?: TChild;
  parent?: string;
}

/** field that we will use to add ad-hoc information to a node
 * this field will be added to each node during parsing the raw data from the api
 */
export interface MetaField {
  meta: { selected: boolean };
}

/** single node description after coming in from the api */
export type RawHierarchySingleNode = HierarchySingleNode<RawHierarchySingleNodeMap>;

/** single node description after our initial custom parsing in preparation of
 * building the tree model
 */
export type ParsedHierarchySingleNode = HierarchySingleNode<ParsedHierarchySingleNode[]> &
  MetaField;

/** in the opensrp api hierarchy response, the raw hierarchy will be key'd
 * by the node's id
 */
export interface RawHierarchySingleNodeMap {
  [id: string]: RawHierarchySingleNode;
}

/** describes the full api Response (raw opensrp hierarchy) */
export interface RawOpenSRPHierarchy {
  locationsHierarchy: {
    map: RawHierarchySingleNodeMap;
    parentChildren: Dictionary<string[]>;
  };
}

/** helper type, shortened form */
export type TreeNode = TreeModel.Node<ParsedHierarchySingleNode>;

export type AutoSelectCallback = (node: TreeNode) => boolean;
