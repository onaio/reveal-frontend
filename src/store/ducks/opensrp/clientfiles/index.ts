import { keyBy, values } from 'lodash';
import { AnyAction, Store } from 'redux';
import { createSelector } from 'reselect';
import SeamlessImmutable from 'seamless-immutable';

/** The reducer name */
export const reducerName = 'clientfiles';

/** Interface for file json object */
export interface File {
  identifier: string; // the unique identifier of the file
  fileLength?: number; // the length of the file (lines / rows)
  fileName: string; // the name of the file
  fileSize?: string; // the of the file
  uploadDate: string; // the date of the latest file update
  providerID: string; // the username of the file creator
  url: string; // download location of the file
}

// actions

/** action type for fetching clients */
export const FILES_FETCHED = 'opensrp/reducer/clients/FILES_FETCHED';
/** action type for removing clients */
export const REMOVE_FILES = 'opensrp/reducer/clients/REMOVE_FILES';

/** interface action to add Files to store */
export interface FetchFilesAction extends AnyAction {
  overwrite: boolean;
  filesById: { [key: string]: File };
  type: typeof FILES_FETCHED;
}

/** Interface for removeFilesAction */
export interface RemoveFilesAction extends AnyAction {
  filesById: {};
  type: typeof REMOVE_FILES;
}

/** Create type for file reducer actions */
export type FileActionTypes = FetchFilesAction | RemoveFilesAction | AnyAction;

// action Creators

/** Fetch files action creator
 * @param {File[]} clientsList - clients array to add to store
 * @param {boolean} overwrite - whether to replace the records in store for clients
 * @return {FetchFileAction} - an action to add clients to redux store
 */
export const fetchFiles = (
  filesList: File[] = [],
  overwrite: boolean = false
): FetchFilesAction => ({
  filesById: keyBy(filesList, (file: File) => file.identifier),
  overwrite,
  type: FILES_FETCHED,
});

// actions

/** removeFilesAction action */
export const removeFilesAction = {
  filesById: {},
  type: REMOVE_FILES,
};

// The reducer

/** interface for files state in redux store */
export interface FileState {
  filesById: { [key: string]: File } | {};
}

/** Create an immutable Files state */
export type ImmutableFileState = FileState & SeamlessImmutable.ImmutableObject<FileState>;

/** initial practitioners-state state */
export const initialState: ImmutableFileState = SeamlessImmutable({
  filesById: {},
});

/** the Files reducer function */
export default function reducer(
  state: ImmutableFileState = initialState,
  action: FileActionTypes
): ImmutableFileState {
  switch (action.type) {
    case FILES_FETCHED:
      const filesToPut = action.overwrite
        ? { ...action.filesById }
        : { ...state.filesById, ...action.filesById };
      return SeamlessImmutable({
        ...state,
        filesById: filesToPut,
      });
    case REMOVE_FILES: {
      return SeamlessImmutable({
        ...state,
        filesById: action.filesById,
      });
    }
    default:
      return state;
  }
}

// Selectors

/** Get all Files keyed by File.identifier
 * @param {Partial<Store>} state - Portion of the store
 * @returns {[key: string]: File} filesById
 */
export const getFilesById = (state: Partial<Store>): { [key: string]: File } => {
  return (state as any)[reducerName].filesById;
};

/** Get all Files as an array of File objects
 * @param {Partial<Store>} state - the redux store
 * @returns {File[]} - an array of File objects
 */
export const getFilesArray = (state: Partial<Store>): File[] => {
  return values(getFilesById(state) || {});
};

/** This interface represents the structure of files filter options/params */
export interface FileFilters {
  fileName?: string /** file name */;
}

/**
 * Gets file name from Filters
 * @param state - the redux store
 * @param props - the file filters object
 */
export const getFileName = (_: Partial<Store>, props: FileFilters) => props.fileName;

/**
 * Gets an array of files filtered by name
 * @param {Partial<Store>} state - the redux store
 * @param {FileFilters} props - the  files filters object
 */
export const getFilesArrayByName = () =>
  createSelector([getFilesArray, getFileName], (files, name) =>
    name ? files.filter(file => file.fileName?.toLowerCase().includes(name.toLowerCase())) : files
  );

/**
 * Returns a selector that gets an array of file objects filtered by one or all
 * of the following:
 *    - file_name
 *
 * These filter params are all optional and are supplied via the prop parameter.
 *
 * This selector is meant to be a memoized replacement for getFilesById.
 *
 * To use this selector, do something like:
 *    const filesArraySelector = makeFilesArraySelector();
 *
 * @param {Partial<Store>} state - the redux store
 * @param {FileFilters} props - the files filters object
 */
export const makeFilesArraySelector = () => {
  return createSelector([getFilesArrayByName()], file1 => file1);
};
