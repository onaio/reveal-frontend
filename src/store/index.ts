import { store } from '@opensrp/store';

export default store;
