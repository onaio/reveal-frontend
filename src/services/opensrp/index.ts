import { IncomingHttpHeaders } from 'http';
import { OPENSRP_API_BASE_URL } from '../../configs/env';
import store from '../../store';
import { getAccessToken } from '../../store/selectors';

/** allowed http methods */
type HTTPMethod = 'GET' | 'POST' | 'PUT';

/** get default HTTP headers for OpenSRP service */
export function getDefaultHeaders(
  accept: string = 'application/json',
  authorizationType: string = 'Bearer',
  contentType: string = 'application/json;charset=UTF-8'
): IncomingHttpHeaders {
  return {
    accept,
    authorization: `${authorizationType} ${getAccessToken(store.getState())}`,
    'content-type': contentType,
  };
}

/** interface to describe URL params object */
export interface URLParams {
  [key: string]: string | number | boolean;
}

/** params option type */
type paramsType = URLParams | null;

/** converts URL params object to string
 * @param {URLParams} obj - the object representing URL params
 * @returns {string} URL params as a string
 */
export function getURLParams(obj: URLParams | {}): string {
  return Object.entries(obj)
    .map(([key, val]) => `${key}=${val}`)
    .join('&');
}

/** converts filter params object to string
 * @param {URLParams} obj - the object representing filter params
 * @returns {string} filter params as a string
 */
export function getFilterParams(obj: URLParams | {}): string {
  return Object.entries(obj)
    .map(([key, val]) => `${key}:${val}`)
    .join(',');
}

/** get payload for fetch
 * @param {HTTPMethod} method - the HTTP method
 * @returns the payload
 */
export function getPayload(method: HTTPMethod) {
  return {
    headers: getDefaultHeaders() as HeadersInit,
    method,
  };
}

/** Get URL
 * @param {string} url - the url
 * @param {paramType} params - the url params object
 * @returns {string} the final url
 */
export function getURL(url: string, params: paramsType = null): string {
  let result = url;
  if (params) {
    result = `${result}?${getURLParams(params)}`;
  }
  return result;
}

/** The OpenSRP service class
 *
 * Sample usage:
 * -------------
 * const service = new OpenSRPService('the-endpoint');
 *
 * **To list all objects**: service.list()
 *
 * **To get one object**: service.read('the-object-identifier')
 *
 * **To create a new object**: service.create(theObject)
 *
 * **To update an object**: service.update(theObject)
 */
export class OpenSRPService {
  public baseURL: string;
  public endpoint: string;
  public generalURL: string;

  constructor(endpoint: string, baseURL: string = OPENSRP_API_BASE_URL) {
    this.endpoint = endpoint;
    this.baseURL = baseURL;
    this.generalURL = `${this.baseURL}${this.endpoint}`;
  }

  /** create method
   * Send a POST request to the general endpoint containing the new object data
   * Successful requests will result in a HTTP status 201 response with no body
   * @param {T} data - the data to be POSTed
   * @param {params} params - the url params object
   * @param {HTTPMethod} method - the HTTP method
   * @returns the object returned by API
   */
  public async create<T>(data: T, params: paramsType = null, method: HTTPMethod = 'POST') {
    const url = getURL(this.generalURL, params);
    const payload = {
      ...getPayload(method),
      'Cache-Control': 'no-cache',
      Pragma: 'no-cache',
      body: JSON.stringify(data),
    };
    const response = await fetch(url, payload);

    if (!response.ok || response.status !== 201) {
      throw new Error(
        `OpenSRPService create on ${this.endpoint} failed, HTTP status ${response.status}`
      );
    }

    return {};
  }

  /** read method
   * Send a GET request to the url for the specific object
   * @param {string|number} id - the identifier of the object
   * @param {params} params - the url params object
   * @param {HTTPMethod} method - the HTTP method
   * @returns the object returned by API
   */
  public async read(id: string | number, params: paramsType = null, method: HTTPMethod = 'GET') {
    const url = getURL(`${this.generalURL}/${id}`, params);
    const response = await fetch(url, getPayload(method));

    if (!response.ok) {
      throw new Error(
        `OpenSRPService read on ${this.endpoint} failed, HTTP status ${response.status}`
      );
    }

    return await response.json();
  }

  /** update method
   * Simply send the updated object as PUT request to the general endpoint URL
   * Successful requests will result in a HTTP status 200/201 response with no body
   * @param {T} data - the data to be POSTed
   * @param {params} params - the url params object
   * @param {HTTPMethod} method - the HTTP method
   * @returns the object returned by API
   */
  public async update<T>(data: T, params: paramsType = null, method: HTTPMethod = 'PUT') {
    const url = getURL(this.generalURL, params);
    const payload = {
      ...getPayload(method),
      'Cache-Control': 'no-cache',
      Pragma: 'no-cache',
      body: JSON.stringify(data),
    };
    const response = await fetch(url, payload);

    if (!response.ok) {
      throw new Error(
        `OpenSRPService update on ${this.endpoint} failed, HTTP status ${response.status}`
      );
    }

    return {};
  }

  /** list method
   * Send a GET request to the general API endpoint
   * @param {params} params - the url params object
   * @param {HTTPMethod} method - the HTTP method
   * @returns list of objects returned by API
   */
  public async list(params: paramsType = null, method: HTTPMethod = 'GET') {
    const url = getURL(this.generalURL, params);
    const response = await fetch(url, getPayload(method));

    if (!response.ok) {
      throw new Error(
        `OpenSRPService list on ${this.endpoint} failed, HTTP status ${response.status}`
      );
    }

    return await response.json();
  }
}
