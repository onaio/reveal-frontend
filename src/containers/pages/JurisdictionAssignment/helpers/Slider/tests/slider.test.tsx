import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { cloneDeep } from 'lodash';
import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedJurisdictionSelectionsSlider } from '..';
import store from '../../../../../../store';
import { fetchTree } from '../../../../../../store/ducks/opensrp/hierarchies';
import { sampleHierarchy } from '../../../../../../store/ducks/opensrp/hierarchies/tests/fixtures';
import { fetchJurisdictionsMetadata } from '../../../../../../store/ducks/opensrp/jurisdictionsMetadata';
import { plans } from '../../../../../../store/ducks/opensrp/PlanDefinition/tests/fixtures';
import { jurisdictionsMetadataArray } from '../../../../../../store/ducks/tests/fixtures';
/** tests for slider view
 * : need to know that changes on the slider are rendered on the ui
 * : need to know that changes on the slider cause the correct changes in the backend
 */

describe('JurisdictionAssignment/Slider', () => {
  it('slider works correctly', () => {
    store.dispatch(fetchTree(sampleHierarchy));
    // prepare fixtures jurisdiction metadata so that it  works with mock hierarchy
    // it will be so that we select one leaf node in sampleHierarchy i.e Akros_1 3951
    const metaData = cloneDeep(jurisdictionsMetadataArray);
    metaData[0].key = '3951';

    store.dispatch(fetchJurisdictionsMetadata(metaData));

    const props = {
      plan: plans[1],
      rootJurisdictionId: '2942',
    };

    const wrapper = mount(
      <Provider store={store}>
        <ConnectedJurisdictionSelectionsSlider {...props} />
      </Provider>
    );

    wrapper.update();

    // Akros_1 is selected by default during mount since it has a threshold of 80 > 0
    expect(wrapper.find('div.slider-section').text()).toMatchInlineSnapshot(
      `"Risk Label  0%00100"`
    );
    expect(wrapper.find('div.info-section').text()).toMatchInlineSnapshot(
      `"NUMBER OF STRUCTURES IN SELECTED JURISDICTIONS159"`
    );

    // simulate a change on the input slider. will set a thresh-hold value
    // that will not select any of the jurisdictions
    expect(toJson(wrapper)).toMatchSnapshot('asdfasdf');

    /** we really should not be doing this, but I am currently unable to simulate
     * the change event
     */
    (wrapper.find('InputRange').props() as any).onChange(81);

    // the risk value should be now different 2
    expect(wrapper.find('.risk-label').text()).toMatchInlineSnapshot(`"81%"`);

    // invoke the method that now actually causes the auto-selection to happen
    (wrapper.find('InputRange').props() as any).onChangeComplete(81);

    // the structure count should now be zero
    expect(wrapper.find('div.info-section').text()).toMatchInlineSnapshot(
      `"NUMBER OF STRUCTURES IN SELECTED JURISDICTIONS0"`
    );
  });
});
