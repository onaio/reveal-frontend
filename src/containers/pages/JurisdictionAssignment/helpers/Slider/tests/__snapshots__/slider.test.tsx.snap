// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JurisdictionAssignment/Slider slider works correctly: asdfasdf 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(JurisdictionSelectionsSlider)
    plan={
      Object {
        "action": Array [
          Object {
            "code": "BCC",
            "description": "Perform BCC for the operational area",
            "goalId": "BCC_complete",
            "identifier": "3f2eef38-38fe-4108-abb3-4e896b880302",
            "prefix": 1,
            "reason": "Routine",
            "subjectCodableConcept": Object {
              "text": "Operational_Area",
            },
            "taskTemplate": "Action1_Perform_BCC",
            "timingPeriod": Object {
              "end": "2019-07-30",
              "start": "2019-07-10",
            },
            "title": "Perform BCC",
          },
          Object {
            "code": "IRS",
            "description": "Visit each structure in the operational area and attempt to spray",
            "goalId": "90_percent_of_structures_sprayed",
            "identifier": "95a5a00f-a411-4fe5-bd9c-460a856239c9",
            "prefix": 2,
            "reason": "Routine",
            "subjectCodableConcept": Object {
              "text": "Residential_Structure",
            },
            "taskTemplate": "Action2_Spray_Structures",
            "timingPeriod": Object {
              "end": "2019-07-30",
              "start": "2019-07-10",
            },
            "title": "Spray Structures",
          },
        ],
        "date": "2019-07-10",
        "effectivePeriod": Object {
          "end": "2019-07-30",
          "start": "2019-07-10",
        },
        "goal": Array [
          Object {
            "description": "Complete BCC for the operational area",
            "id": "BCC_complete",
            "priority": "medium-priority",
            "target": Array [
              Object {
                "detail": Object {
                  "detailQuantity": Object {
                    "comparator": ">=",
                    "unit": "each",
                    "value": 1,
                  },
                },
                "due": "2019-07-10",
                "measure": "Number of BCC communication activities that happened",
              },
            ],
          },
          Object {
            "description": "Spray 90 % of structures in the operational area",
            "id": "90_percent_of_structures_sprayed",
            "priority": "medium-priority",
            "target": Array [
              Object {
                "detail": Object {
                  "detailQuantity": Object {
                    "comparator": ">=",
                    "unit": "percent",
                    "value": 90,
                  },
                },
                "due": "2019-07-30",
                "measure": "Percent of structures sprayed",
              },
            ],
          },
        ],
        "identifier": "8fa7eb32-99d7-4b49-8332-9ecedd6d51ae",
        "jurisdiction": Array [
          Object {
            "code": "35968df5-f335-44ae-8ae5-25804caa2d86",
          },
          Object {
            "code": "3952",
          },
          Object {
            "code": "ac7ba751-35e8-4b46-9e53-3cbaad193697",
          },
        ],
        "name": "TwoTwoOne_01_IRS_2019-07-10",
        "serverVersion": 1563303151426,
        "status": "active",
        "title": "TwoTwoOne_01 IRS 2019-07-10",
        "useContext": Array [
          Object {
            "code": "interventionType",
            "valueCodableConcept": "IRS",
          },
        ],
        "version": "1",
      }
    }
    rootJurisdictionId="2942"
  >
    <JurisdictionSelectionsSlider
      autoSelectCreator={[Function]}
      fetchJurisdictionsMetadataCreator={[Function]}
      fetchTreeCreator={[Function]}
      jurisdictionsMetadata={
        Array [
          Object {
            "description": "Jurisdiction Metadata for test1 id 79b139c-3a20-4656-b684-d2d9ed83c94e",
            "documentId": "bc15c480-0319-4711-8bfe-6e0a64b297e3",
            "key": "3951",
            "label": "test1 metadata",
            "locationId": "",
            "providerId": "onatest",
            "resolveSettings": false,
            "serverVersion": 1592833111093,
            "settingIdentifier": "jurisdiction_metadata-risk",
            "settingMetadataId": "92",
            "settingsId": "33",
            "teamId": "",
            "uuid": "e56ce640-38ee-444a-9f06-a53d64a96116",
            "v1Settings": false,
            "value": "80",
          },
          Object {
            "description": "Jurisdiction Metadata for test2 id 02ebbc84-5e29-4cd5-9b79-c594058923e9",
            "documentId": "bc15c480-0319-4711-8bfe-6e0a64b297e3",
            "key": "02ebbc84-5e29-4cd5-9b79-c594058923e9",
            "label": "test2 metadata",
            "locationId": "",
            "providerId": "onatest",
            "resolveSettings": false,
            "serverVersion": 1592833111093,
            "settingIdentifier": "jurisdiction_metadata-risk",
            "settingMetadataId": "93",
            "settingsId": "33",
            "teamId": "",
            "uuid": "fd8dfc3e-7202-4ceb-97ce-13c0b7966e61",
            "v1Settings": false,
            "value": "70",
          },
        ]
      }
      onClickNext={[Function]}
      plan={
        Object {
          "action": Array [
            Object {
              "code": "BCC",
              "description": "Perform BCC for the operational area",
              "goalId": "BCC_complete",
              "identifier": "3f2eef38-38fe-4108-abb3-4e896b880302",
              "prefix": 1,
              "reason": "Routine",
              "subjectCodableConcept": Object {
                "text": "Operational_Area",
              },
              "taskTemplate": "Action1_Perform_BCC",
              "timingPeriod": Object {
                "end": "2019-07-30",
                "start": "2019-07-10",
              },
              "title": "Perform BCC",
            },
            Object {
              "code": "IRS",
              "description": "Visit each structure in the operational area and attempt to spray",
              "goalId": "90_percent_of_structures_sprayed",
              "identifier": "95a5a00f-a411-4fe5-bd9c-460a856239c9",
              "prefix": 2,
              "reason": "Routine",
              "subjectCodableConcept": Object {
                "text": "Residential_Structure",
              },
              "taskTemplate": "Action2_Spray_Structures",
              "timingPeriod": Object {
                "end": "2019-07-30",
                "start": "2019-07-10",
              },
              "title": "Spray Structures",
            },
          ],
          "date": "2019-07-10",
          "effectivePeriod": Object {
            "end": "2019-07-30",
            "start": "2019-07-10",
          },
          "goal": Array [
            Object {
              "description": "Complete BCC for the operational area",
              "id": "BCC_complete",
              "priority": "medium-priority",
              "target": Array [
                Object {
                  "detail": Object {
                    "detailQuantity": Object {
                      "comparator": ">=",
                      "unit": "each",
                      "value": 1,
                    },
                  },
                  "due": "2019-07-10",
                  "measure": "Number of BCC communication activities that happened",
                },
              ],
            },
            Object {
              "description": "Spray 90 % of structures in the operational area",
              "id": "90_percent_of_structures_sprayed",
              "priority": "medium-priority",
              "target": Array [
                Object {
                  "detail": Object {
                    "detailQuantity": Object {
                      "comparator": ">=",
                      "unit": "percent",
                      "value": 90,
                    },
                  },
                  "due": "2019-07-30",
                  "measure": "Percent of structures sprayed",
                },
              ],
            },
          ],
          "identifier": "8fa7eb32-99d7-4b49-8332-9ecedd6d51ae",
          "jurisdiction": Array [
            Object {
              "code": "35968df5-f335-44ae-8ae5-25804caa2d86",
            },
            Object {
              "code": "3952",
            },
            Object {
              "code": "ac7ba751-35e8-4b46-9e53-3cbaad193697",
            },
          ],
          "name": "TwoTwoOne_01_IRS_2019-07-10",
          "serverVersion": 1563303151426,
          "status": "active",
          "title": "TwoTwoOne_01 IRS 2019-07-10",
          "useContext": Array [
            Object {
              "code": "interventionType",
              "valueCodableConcept": "IRS",
            },
          ],
          "version": "1",
        }
      }
      rootJurisdictionId="2942"
      structuresCount={159}
      tree={
        Node {
          "children": Array [
            Node {
              "children": Array [
                Node {
                  "children": Array [],
                  "config": Object {
                    "childrenPropertyName": "children",
                    "modelComparatorFn": undefined,
                  },
                  "model": Object {
                    "id": "3951",
                    "label": "Akros_1",
                    "meta": Object {
                      "selected": true,
                      "selectedBy": "Auto-selection",
                    },
                    "node": Object {
                      "attributes": Object {
                        "geographicLevel": 2,
                        "structureCount": 159,
                      },
                      "locationId": "3951",
                      "name": "Akros_1",
                      "parentLocation": Object {
                        "locationId": "3019",
                        "voided": false,
                      },
                      "voided": false,
                    },
                    "parent": "3019",
                  },
                  "parent": [Circular],
                },
              ],
              "config": Object {
                "childrenPropertyName": "children",
                "modelComparatorFn": undefined,
              },
              "model": Object {
                "children": Array [
                  Object {
                    "id": "3951",
                    "label": "Akros_1",
                    "meta": Object {
                      "selected": true,
                      "selectedBy": "Auto-selection",
                    },
                    "node": Object {
                      "attributes": Object {
                        "geographicLevel": 2,
                        "structureCount": 159,
                      },
                      "locationId": "3951",
                      "name": "Akros_1",
                      "parentLocation": Object {
                        "locationId": "3019",
                        "voided": false,
                      },
                      "voided": false,
                    },
                    "parent": "3019",
                  },
                ],
                "id": "3019",
                "label": "Mtendere",
                "meta": Object {
                  "selected": true,
                  "selectedBy": "Auto-selection",
                },
                "node": Object {
                  "attributes": Object {
                    "geographicLevel": 1,
                    "structureCount": 1,
                  },
                  "locationId": "3019",
                  "name": "Mtendere",
                  "parentLocation": Object {
                    "locationId": "2942",
                    "voided": false,
                  },
                  "voided": false,
                },
                "parent": "2942",
              },
              "parent": [Circular],
            },
          ],
          "config": Object {
            "childrenPropertyName": "children",
            "modelComparatorFn": undefined,
          },
          "model": Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "id": "3951",
                    "label": "Akros_1",
                    "meta": Object {
                      "selected": true,
                      "selectedBy": "Auto-selection",
                    },
                    "node": Object {
                      "attributes": Object {
                        "geographicLevel": 2,
                        "structureCount": 159,
                      },
                      "locationId": "3951",
                      "name": "Akros_1",
                      "parentLocation": Object {
                        "locationId": "3019",
                        "voided": false,
                      },
                      "voided": false,
                    },
                    "parent": "3019",
                  },
                ],
                "id": "3019",
                "label": "Mtendere",
                "meta": Object {
                  "selected": true,
                  "selectedBy": "Auto-selection",
                },
                "node": Object {
                  "attributes": Object {
                    "geographicLevel": 1,
                    "structureCount": 1,
                  },
                  "locationId": "3019",
                  "name": "Mtendere",
                  "parentLocation": Object {
                    "locationId": "2942",
                    "voided": false,
                  },
                  "voided": false,
                },
                "parent": "2942",
              },
            ],
            "id": "2942",
            "label": "Lusaka",
            "meta": Object {
              "selected": true,
              "selectedBy": "Auto-selection",
            },
            "node": Object {
              "attributes": Object {
                "geographicLevel": 0,
              },
              "locationId": "2942",
              "name": "Lusaka",
              "voided": false,
            },
          },
        }
      }
    >
      <div
        className="mt-3"
      >
        <hr />
        <Row
          tag="div"
        >
          <div
            className="row"
          >
            <Col
              md={
                Object {
                  "offset": 2,
                  "size": 8,
                }
              }
              style={
                Object {
                  "textAlign": "center",
                }
              }
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="12"
            >
              <div
                className="col-12 col-md-8 offset-md-2"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <h4
                  className="mb-5 font-weight-bold"
                >
                  Adjust slider to auto-target jurisdictions
                </h4>
                <Row
                  className="auto-target-row mb-3"
                  tag="div"
                >
                  <div
                    className="auto-target-row mb-3 row"
                  >
                    <Col
                      className="slider-section py-3 px-5"
                      md={6}
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                      xs="12"
                    >
                      <div
                        className="slider-section py-3 px-5 col-12 col-md-6"
                      >
                        <p>
                          Risk Label
                            
                          <span
                            className="risk-label"
                          >
                            0%
                          </span>
                        </p>
                        <InputRange
                          allowSameValues={false}
                          classNames={
                            Object {
                              "activeTrack": "input-range__track input-range__track--active",
                              "disabledInputRange": "input-range input-range--disabled",
                              "inputRange": "input-range",
                              "labelContainer": "input-range__label-container",
                              "maxLabel": "input-range__label input-range__label--max",
                              "minLabel": "input-range__label input-range__label--min",
                              "slider": "input-range__slider",
                              "sliderContainer": "input-range__slider-container",
                              "track": "input-range__track input-range__track--background",
                              "valueLabel": "input-range__label input-range__label--value",
                            }
                          }
                          disabled={false}
                          maxValue={100}
                          minValue={0}
                          onChange={[Function]}
                          onChangeComplete={[Function]}
                          step={1}
                          value={0}
                        >
                          <div
                            aria-disabled={false}
                            className="input-range"
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onTouchStart={[Function]}
                          >
                            <Label
                              classNames={
                                Object {
                                  "activeTrack": "input-range__track input-range__track--active",
                                  "disabledInputRange": "input-range input-range--disabled",
                                  "inputRange": "input-range",
                                  "labelContainer": "input-range__label-container",
                                  "maxLabel": "input-range__label input-range__label--max",
                                  "minLabel": "input-range__label input-range__label--min",
                                  "slider": "input-range__slider",
                                  "sliderContainer": "input-range__slider-container",
                                  "track": "input-range__track input-range__track--background",
                                  "valueLabel": "input-range__label input-range__label--value",
                                }
                              }
                              type="min"
                            >
                              <span
                                className="input-range__label input-range__label--min"
                              >
                                <span
                                  className="input-range__label-container"
                                >
                                  0
                                </span>
                              </span>
                            </Label>
                            <Track
                              classNames={
                                Object {
                                  "activeTrack": "input-range__track input-range__track--active",
                                  "disabledInputRange": "input-range input-range--disabled",
                                  "inputRange": "input-range",
                                  "labelContainer": "input-range__label-container",
                                  "maxLabel": "input-range__label input-range__label--max",
                                  "minLabel": "input-range__label input-range__label--min",
                                  "slider": "input-range__slider",
                                  "sliderContainer": "input-range__slider-container",
                                  "track": "input-range__track input-range__track--background",
                                  "valueLabel": "input-range__label input-range__label--value",
                                }
                              }
                              onTrackDrag={[Function]}
                              onTrackMouseDown={[Function]}
                              percentages={
                                Object {
                                  "max": 0,
                                  "min": 0,
                                }
                              }
                            >
                              <div
                                className="input-range__track input-range__track--background"
                                onMouseDown={[Function]}
                                onTouchStart={[Function]}
                              >
                                <div
                                  className="input-range__track input-range__track--active"
                                  style={
                                    Object {
                                      "left": "0%",
                                      "width": "0%",
                                    }
                                  }
                                />
                                <Slider
                                  classNames={
                                    Object {
                                      "activeTrack": "input-range__track input-range__track--active",
                                      "disabledInputRange": "input-range input-range--disabled",
                                      "inputRange": "input-range",
                                      "labelContainer": "input-range__label-container",
                                      "maxLabel": "input-range__label input-range__label--max",
                                      "minLabel": "input-range__label input-range__label--min",
                                      "slider": "input-range__slider",
                                      "sliderContainer": "input-range__slider-container",
                                      "track": "input-range__track input-range__track--background",
                                      "valueLabel": "input-range__label input-range__label--value",
                                    }
                                  }
                                  key="max"
                                  maxValue={100}
                                  minValue={0}
                                  onSliderDrag={[Function]}
                                  onSliderKeyDown={[Function]}
                                  percentage={0}
                                  type="max"
                                  value={0}
                                >
                                  <span
                                    className="input-range__slider-container"
                                    style={
                                      Object {
                                        "left": "0%",
                                        "position": "absolute",
                                      }
                                    }
                                  >
                                    <Label
                                      classNames={
                                        Object {
                                          "activeTrack": "input-range__track input-range__track--active",
                                          "disabledInputRange": "input-range input-range--disabled",
                                          "inputRange": "input-range",
                                          "labelContainer": "input-range__label-container",
                                          "maxLabel": "input-range__label input-range__label--max",
                                          "minLabel": "input-range__label input-range__label--min",
                                          "slider": "input-range__slider",
                                          "sliderContainer": "input-range__slider-container",
                                          "track": "input-range__track input-range__track--background",
                                          "valueLabel": "input-range__label input-range__label--value",
                                        }
                                      }
                                      type="value"
                                    >
                                      <span
                                        className="input-range__label input-range__label--value"
                                      >
                                        <span
                                          className="input-range__label-container"
                                        >
                                          0
                                        </span>
                                      </span>
                                    </Label>
                                    <div
                                      aria-valuemax={100}
                                      aria-valuemin={0}
                                      aria-valuenow={0}
                                      className="input-range__slider"
                                      draggable="false"
                                      onKeyDown={[Function]}
                                      onMouseDown={[Function]}
                                      onTouchStart={[Function]}
                                      role="slider"
                                      tabIndex="0"
                                    />
                                  </span>
                                </Slider>
                              </div>
                            </Track>
                            <Label
                              classNames={
                                Object {
                                  "activeTrack": "input-range__track input-range__track--active",
                                  "disabledInputRange": "input-range input-range--disabled",
                                  "inputRange": "input-range",
                                  "labelContainer": "input-range__label-container",
                                  "maxLabel": "input-range__label input-range__label--max",
                                  "minLabel": "input-range__label input-range__label--min",
                                  "slider": "input-range__slider",
                                  "sliderContainer": "input-range__slider-container",
                                  "track": "input-range__track input-range__track--background",
                                  "valueLabel": "input-range__label input-range__label--value",
                                }
                              }
                              type="max"
                            >
                              <span
                                className="input-range__label input-range__label--max"
                              >
                                <span
                                  className="input-range__label-container"
                                >
                                  100
                                </span>
                              </span>
                            </Label>
                          </div>
                        </InputRange>
                      </div>
                    </Col>
                    <Col
                      className="info-section p-3"
                      md={6}
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                      xs="12"
                    >
                      <div
                        className="info-section p-3 col-12 col-md-6"
                      >
                        <p>
                          <FontAwesomeIcon
                            icon="home"
                          >
                            <i
                              className="fa"
                            />
                          </FontAwesomeIcon>
                        </p>
                        <p>
                          NUMBER OF STRUCTURES IN SELECTED JURISDICTIONS
                        </p>
                        <p>
                          159
                        </p>
                      </div>
                    </Col>
                  </div>
                </Row>
              </div>
            </Col>
          </div>
        </Row>
        <hr />
        <Button
          className="btn btn-success float-right mt-3"
          color="secondary"
          onClick={[Function]}
          tag="button"
        >
          <button
            aria-label={null}
            className="btn btn-success float-right mt-3 btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            Continue to next step
          </button>
        </Button>
      </div>
    </JurisdictionSelectionsSlider>
  </Connect(JurisdictionSelectionsSlider)>
</Provider>
`;
